networkoperator/v1/001-egressrouter-crd.yaml

```diff
@@ -0,0 +1,231 @@
+                          format: int32
+                          maximum: 65535
+                          minimum: 1
+                        protocol:
+                          description: Protocol can be TCP, SCTP or UDP.
+                          type: string
+                          enum:
+                          - TCP
+                          - UDP
+                          - SCTP
+                        targetPort:
+                          description: TargetPort allows specifying the port number
+                            on the remote destination to which the traffic gets redirected
+                            to. If unspecified, the value from "Port" is used.
+                          type: integer
+                          format: int32
+                          maximum: 65535
+                          minimum: 1
+          status:
+            description: Observed status of EgressRouter.
+            type: object
+            properties:
+              conditions:
+                description: Observed status of the egress router
+                type: array
+                items:
+                  description: EgressRouterStatusCondition represents the state of
+                    the egress router's managed and monitored components.
+                  type: object
+                  required:
+                  - lastTransitionTime
+                  - status
+                  - type
+                  properties:
+                    lastTransitionTime:
+                      description: LastTransitionTime is the time of the last update
+                        to the current status property.
+                      type: string
+                      format: date-time
+                    message:
+                      description: Message provides additional information about the
+                        current condition. This is only to be consumed by humans.  It
+                        may contain Line Feed characters (U+000A), which should be
+                        rendered as new lines.
+                      type: string
+                    reason:
+                      description: Reason is the CamelCase reason for the condition's
+                        current status.
+                      type: string
+                    status:
+                      description: Status of the condition, one of True, False, Unknown.
+                      type: string
+                      enum:
+                      - "True"
+                      - "False"
+                      - Unknown
+                    type:
+                      description: Type specifies the aspect reported by this condition;
+                        one of Available, Progressing, Degraded
+                      type: string
+                      enum:
+                      - Available
+                      - Progressing
+                      - Degraded
+    served: true
+    storage: true
+    subresources: {}
```

Comment:
no status subresource?

Reply:
hmm, if I set that, then any updates to egressrouterspec won't get relected to the status right ? https://master.book.kubebuilder.io/reference/generating-crd.html?highlight=subres#subresources and I guess we want the status to reflect the changes right ? Or am I wrong in understanding what status as a subresource mean?

Reply:
You will have two update methods: Update for spec and UpdateStatus for status. Reason is that usually the operator writes only to status and the admin writes the spec. So you can split it with different RBAC rules.

Important: you will not be able to change this easily lateron. Normally it is a good idea to enable the status subresource.